### PR TITLE
UTY-963: Core C# SDK Docs Changes

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Updates all the docs to reflect the changes to the core. Some things have been omitted for now (reference type providers).
#### Tests
N/A
#### Documentation
yes
#### Primary reviewers
@ElleEss

### CODE REVIEW

File: docs/content/authority.md [hunk 1]
```diff
@@ -72,11 +72,9 @@ This means entities _may_ initially have `NotAuthoritative<T>` attached for a ti
 
 ### What happens when a worker instance gains or loses authority over a SpatialOS entity
 
-When authority changes over a SpatialOS component the Unity GDK automatically adds an `AuthoritiesChanged<T>` reactive ECS component to the corresponding ECS entity (for `T`, the component that changed authority).
+When authority changes over a SpatialOS component the Unity GDK automatically adds an `AuthorityChanges<T>` reactive ECS component to the corresponding ECS entity (for `T`, the component that changed authority).
```
Comment (jared-improbable) [inline docs/content/authority.md]: "for `T`" -> "where `T` is"

File: docs/content/events.md [hunk 2]
```diff
@@ -13,9 +13,9 @@ Events are for broadcasting information between worker instances about a transie
 
 ### Sending events
 
-A worker instance can send an event using a `EventSender<T>` ECS component (where `T` is the SpatialOS component that the event is defined in).
+A worker instance can send an event using a `ComponentName.EventSenders.EventName` ECS component (where `ComponentName` is the name of the component that the event is defined in and `EventName` is the name of the event in schema).
```
Comment (jared-improbable) [inline docs/content/events.md]: Avoid writing useful information as parentheticals. Just make them a full sentence.

File: docs/content/custom-replication-system.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ When a worker sends a SpatialOS event, the Unity GDK puts the event object into
 
 If some ECS components need more complex replication logic, you can create custom replication systems on a per-component basis. To do this:
 
-* Your custom replication system must extend the `CustomSpatialOSSendSystem<T>` class (where `T` is a SpatialOS component). Note that this will disable the standard replication for `T`.
+* Your custom replication system must extend the `CustomSpatialOSSendSystem<T>` class (where `T` is a SpatialOS component). Note that this will disable the standard replication for `T` and ensure the system is ran at the correct point in the update lifecycle.
```
[the remark this answers is not in the document]
Comment (jamiebrynes7) [reply docs/content/custom-replication-system.md]: Will also fix the passive voice in the latter half of the sentence

File: docs/content/entity-checkout-process.md [hunk 4]
```diff
@@ -8,14 +8,14 @@
...
 - An [authority maker tag](authority.md) (`Authoritative<T>`, `NotAuthoritative<T>` or `AuthorityLostImminent<T>`) based on the Worker's component write authority.
 
 In addition, the following components are added to your entity as well:
 - `SpatialEntityId`: Holds the corresponding SpatialOS EntityId of an entity.
 - `NewlyAddedSpatialOSEntity`: Tag component for marking entities that were just checked-out. This component is automatically removed from your entity at the end of the frame it was created.
-- `WorldCommandSender`: Component that exposes the API for [sending world commands](commands.md#world-commands) (e.g. create entity, delete entity).
-- A `CommandRequestSender<T>` for every SpatialOS schema component `T` that defines a command: Component that exposes the API for [sending schema commands](commands.md#sending-command-requests) defined in `T`.
+- World Command Senders: Components that exposes the API for [sending world commands](commands.md#world-commands) (e.g. create entity, delete entity).
```
[the remark this answers is not in the document]
Comment (jamiebrynes7) [reply docs/content/entity-checkout-process.md]: Yes - its multiple components now so it doesn't refer to a singular components.

Tbh - not sure how to phrase this properly

[the remark this answers is not in the document]
Comment (jared-improbable) [reply docs/content/entity-checkout-process.md]: "World command senders: A set of components that expose the APIs for [sending world commands](commands.md#world-commands) (e.g. create entity, delete entity)."

I got rid of the Title Case, since we try to avoid that unless it's actually a name in code.

File: docs/content/custom-replication-system.md [hunk 5]
```diff
@@ -90,20 +95,18 @@ public class TransformSendSystem : CustomSpatialOSSendSystem<SpatialOSTransform>
...
-[UpdateInGroup(typeof(UpdateGroupSpatialOSSend))]
-[UpdateBefore(typeof(UnityEngine.Experimental.PlayerLoop.FixedUpdate))]
 public class CubeColorSendSystem : CustomSpatialOSSendSystem<SpatialOSCubeColor>
 {
     public struct ColorData
     {
         public readonly int Length;
         public ComponentDataArray<SpatialOSCubeColor> CubeColors;
-        public ComponentDataArray<EventSender<SpatialOSCubeColor>> CubeColorEventSenders;
+        public ComponentDataArray<CuberColor.EventSenders.ChangeColor> ChangeColorEventSenders;
```
Comment (jessicafalk) [inline docs/content/custom-replication-system.md]: in our code its `CuberColor.EventSender.ChangeColor` (without the s at the end of  Eventsenders)

File: docs/content/custom-replication-system.md [hunk 6]
```diff
@@ -115,24 +118,24 @@ public class CubeColorSendSystem : CustomSpatialOSSendSystem<SpatialOSCubeColor>
...
-                update.changeColor.Add(global::Generated.Playground.ColorData.ToSpatial(event));
+                var eventObj = eventsObj.AddObject(1); // NOTE: 1 corresponds to the event index in schema. 
+                global::Generated.Playground.ColorData.Serialization.Serialize(event, eventObj)
             }
 
-            SpatialOSCubeColorTranslation.SendComponentUpdate(worker.Connection, entityId, update);
-
-            eventSender.ClearChangeColorEvents();
+            worker.Connection.SendComponentUpdate(entityId, new ComponentUpdate(update));
+            changeColorEvents.Clear();
```
Comment (jessicafalk) [inline docs/content/custom-replication-system.md]: in the `TriggerColorChangeSystem` you don't  need to do all that. You simply add the event to the events list 🤔 
which one should be the proposed way?

File: docs/content/entity-checkout-process.md [hunk 7]
```diff
@@ -8,14 +8,14 @@
...
 
 In addition, the following components are added to your entity as well:
 - `SpatialEntityId`: Holds the corresponding SpatialOS EntityId of an entity.
 - `NewlyAddedSpatialOSEntity`: Tag component for marking entities that were just checked-out. This component is automatically removed from your entity at the end of the frame it was created.
-- `WorldCommandSender`: Component that exposes the API for [sending world commands](commands.md#world-commands) (e.g. create entity, delete entity).
-- A `CommandRequestSender<T>` for every SpatialOS schema component `T` that defines a command: Component that exposes the API for [sending schema commands](commands.md#sending-command-requests) defined in `T`.
+- World command senders: A set of components that expose the API for [sending world commands](commands.md#world-commands) (e.g. create entity, delete entity).
+- A `ComponentName.CommandSenders.CommandName` for every SpatialOS schema command: Component that exposes the API for [sending schema commands](commands.md#sending-command-requests).
```
Comment (jessicafalk) [inline docs/content/entity-checkout-process.md]: also an s too many: should be `CommandSender`

File: docs/content/custom-replication-system.md (hunk 6 above)
Comment (jamiebrynes7) [reply docs/content/custom-replication-system.md]: This isn't an example of sending events normally - this is for if you wanted to write custom serialisation logic for events 😉

File: docs/content/entity-checkout-process.md (hunk 7 above)
Comment (jamiebrynes7) [reply docs/content/entity-checkout-process.md]: It is actually `CommandSenders`. The events and commands are inconsistent 😬  cc @zeroZshadow

Comment (jessicafalk) [reply docs/content/entity-checkout-process.md]: in that case we should make it consistent :P

File: docs/content/custom-replication-system.md (hunk 6 above)
Comment (jessicafalk) [reply docs/content/custom-replication-system.md]: ahh, got it 👍

File: docs/content/custom-replication-system.md [hunk 8]
```diff
@@ -115,24 +118,24 @@ public class CubeColorSendSystem : CustomSpatialOSSendSystem<SpatialOSCubeColor>
...
             }
 
-            var update = new global::Playground.CubeColor.Update();
-
+            var update = new SchemaComponentUpdate(component.ComponentId);
+            var eventsObj = update.GetEvents();
             foreach(var event in changeColorEvents)
             {
-                update.changeColor.Add(global::Generated.Playground.ColorData.ToSpatial(event));
+                var eventObj = eventsObj.AddObject(1); // NOTE: 1 corresponds to the event index in schema.
```
[the remark this answers is not in the document]
Comment (jamiebrynes7) [reply docs/content/custom-replication-system.md]: We can introduce some helpers - but way out of scope for this PR

File: docs/content/entity-checkout-process.md [hunk 9]
```diff
@@ -8,14 +8,14 @@
 For every entity we receive from SpatialOS, an ECS entity representing the SpatialOS entity is automatically created in the Worker's world through the `SpatialOSReceiveSystem`.
 
 For each of your entity's SpatialOS components, the following Unity ECS components are automatically added:
-- `SpatialOS<name of schema component>`: Codegenerated struct of type `ISpatialComponentData` or `Component` for accessing component field values. Fields of these components are automatically kept up-to-date.
+- `SpatialOS<name of schema component>`: Codegenerated struct of type `ISpatialComponentData` for accessing component field values. Fields of these components are automatically kept up-to-date.
```
[the remark this answers is not in the document]
Comment (jamiebrynes7) [reply docs/content/entity-checkout-process.md]: Yeah I can see that being confusing - will have a think about how to represent that better

File: docs/content/custom-replication-system.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (martinzlocha) [reply docs/content/custom-replication-system.md]: Ok, is there a ticket for this?

File: docs/content/commands.md [hunk 10]
```diff
@@ -70,34 +72,37 @@ public class BuildSystem : ComponentSystem
...
+
             
-            requestSender.SendBuildWallRequest(entityId, buildRequest);
+            requestSender.RequestsToSend.Add(request);
         }
     }
 }
 ```
 
 This system runs on a client. It injects all SpatialOS entities in the client's view that have the `Builder` SpatialOS component, and sends a `build_wall` command to each SpatialOS entity (which will be received by the managed worker).
```
Comment (ElleEss) [inline docs/content/commands.md]: I know you didn't edit this but...
(1) It injects what in the SpatialOS entities? 
Or does it inject the SpatialOS entites *into* the client' view?
I can't follow the sentence.
(2) Does it send the `build_wall` command to each SpatialOS entity which has the `Builder` component or all of them. Currently it says "each" which suggests all of them. I suspect it's the one with the components only?
(3) Which managed worker (server-worker) receives the commands?  Or is it just general "all server workers"?
(4) We need to use the current terminology:
"client-worker", "managed-worker".

So something like the following (but my interpretation may be out).

"This system runs on a client-worker. It injects <something into??> all SpatialOS entities currently in the client-worker's view that have the `Builder` SpatialOS component. It then sends a `build_wall` command to each SpatialOS entity with that component. All server-workers receive this command."

File: docs/content/commands.md [hunk 11]
```diff
@@ -70,34 +72,37 @@ public class BuildSystem : ComponentSystem
...
-
-You can use the object in the list itself to respond to that particular request.
+The Unity GDK attaches a `ComponentName.CommandRequests.CommandName` ECS component to the specified ECS entity, where `ComponentName` is the SpatialOS component the command is defined in and `CommandName` is the name of the command in schema. `ComponentName.CommandRequests.CommandName` contains a list of type `ComponentName.CommandName.ReceivedRequest`. The Unity GDK cleans it up at the end of the tick.
 
-> **Note**: `CommandRequests<T>` is a `Component`, not an `IComponentData`. This means you must use a `ComponentArray<T>` for injection rather than a `ComponentDataArray<T>`.
+To respond to the request, use `ComponentName.CommandResponders.CommandName` for that given command. This contains a list of type `ComponentName.CommandName.Response`. Create and add a `ComponentName.CommandName.Response` object to this list and the GDK will automatically send the response for you.
```
Comment (ElleEss) [inline docs/content/commands.md]: No need to say "will automatically" - it's stronger (and suggests a trustworthy, reliable product) to write, "...and the GDK automatically sends the response for you.

File: docs/content/commands.md [hunk 12]
```diff
@@ -117,33 +123,35 @@ public class BuildWallHandlerSystem : ComponentSystem
...
+Like requests, when an ECS entity receives a command response, the Unity GDK attaches a `ComponentName.CommandResponses.CommandName` ECS component to the ECS entity, where `ComponentName` is the SpatialOS component the command is defined in and `CommandName` is the name of the command in schema.
 
-The ECS entity that sent the request receives the response. The response object includes the payload of the response and the request payload that originally send the command. **This payload is null** when no payload is sent back.
+`ComponentName.CommandResponses.CommandName` contains a list of `ComponentName.CommandName.ReceivedResponse`. The `ComponentName.CommandName.ReceivedResponse` includes the payload of the response and the request payload that originally was sent with the command. **This payload is null** when the command fails.
 
-> **Note**: `CommandResponses<T>` is a `Component`, not an `IComponentData`. This means you must use a `ComponentArray<T>` for injection rather than a `ComponentDataArray<T>`.
+The ECS entity that sent the request receives the response. The response object.
```
Comment (ElleEss) [inline docs/content/commands.md]: Not sure what "The response object." means here?

File: docs/content/commands.md [hunk 13]
```diff
@@ -182,43 +190,37 @@ public class BuildWallResponseHandler : ComponentSystem
...
+    * Receiving a response - `WorldCommands.ReserveEntityIds.CommandResponses`. This contains a list of `WorldCommands.ReserveEntityIds.ReceivedResponse` structs. 
+* CreateEntity
+    * Sending a request - `WorldCommands.CreateEntity.CommandSender`. This contains a list of `WorldCommands.CreateEntity.Request` structs. Add a struct to the list to send the command.
+        * `EntityId` and `TimeoutMillis` are optional. 
+        * If you do specify an `EntityId`, you need to get this from a `ReserveEntityIds` command.
+    * Receiving a response - `WorldCommands.CreateEntity.CommandResponses`. This contains a list of `WorldCommands.CreateEntity.ReceivedResponse`. 
+* DeleteEntity
+    * Sending a request - `WorldCommands.DeleteEntity.CommandSender`. This contains a list of `WorldCommands.DeleteEntity.Request` structs. Add a struct to the list to send the command.
+        * `TimeoutMillis` is optional.
+    *Receiving a response - `WorldCommands.DeleteEntity.CommandResponses`. This contains a list of `WorldCommands.DeleteEntity.ReceivedResponse`.
```
Comment (ElleEss) [inline docs/content/commands.md]: Formatting - we need a gap after the star in "*Receiving..."

File: docs/content/component-data.md [hunk 14]
```diff
@@ -9,56 +9,33 @@ SpatialOS components are generated from `.schema` files into components that the
 ### Overview
 
 A `struct`, which implements `Unity.Entities.IComponentData` and `Improbable.Gdk.Core.ISpatialComponentData`,
-is generated for each [blittable](https://docs.microsoft.com/en-us/dotnet/framework/interop/blittable-and-non-blittable-types) SpatialOS component.
-A `class`, which extends `UnityEngine.Component` and implements `Improbable.Gdk.Core.ISpatialComponentData` is generated for each non-blittable SpatialOS component.
+is generated for each SpatialOS component.
```
Comment (ElleEss) [inline docs/content/component-data.md]: Looking at this sentence and the one which precedes it in the doc, which I copy below...

"SpatialOS components are generated from .schema files into components that the Unity ECS can understand. See the schemalang docs for details on how to create schema components."

Neither the first sentence, or the one under `### Overview` tells us how the SpatialOS components are generated. What does the user have to do? What causes this to happen? It reads as if it happens by magic.

File: docs/content/component-data.md [hunk 15]
```diff
@@ -9,56 +9,33 @@ SpatialOS components are generated from `.schema` files into components that the
...
 
 Each of these structs will be named `SpatialOS[schemalang component name]` and will contain only the schema data fields.
```
Comment (ElleEss) [inline docs/content/component-data.md]: This sentence doesn't make it clear who does the naming. Is it the generation process?
Also "will be named"  and "will contain" reads like a prophecy rather than a statement.  
Is "[schemalang component name]" dependent on the component?
Suggest:

" The generation process names each of these structs according to the relevant schemalang component name,` `SpatialOS[schemalang component name]`. The structs only contain the schema data fields."

File: docs/content/component-data.md [hunk 16]
```diff
@@ -9,56 +9,33 @@ SpatialOS components are generated from `.schema` files into components that the
...
 It will **Not** contain any fields or methods pertaining to [commands](commands.md) or [events](events.md) defined on that component.
```
Comment (ElleEss) [inline docs/content/component-data.md]: "It will **Not**...should carry on from the sentence above, with no new para. Also, best to use italics for stress.
Suggest "relating" rather than "pertaining" as it's less formal, more accessible.
Suggest:

"They do _not_ contain any fields or methods relating to [commands](commands.md) or [events](events.md) defined on that component."

File: docs/content/component-data.md [hunk 17]
```diff
@@ -9,56 +9,33 @@ SpatialOS components are generated from `.schema` files into components that the
...
-{
-  public int IntBlittable;
-  public Dictionary<int, float> MapNonBlittable;
-  public string StringNonBlittable;
-}
-```
-
 ### Reading and writing
 
 When a SpatialOS entity is [checked out](entity-checkout-process.md), the generated components will be automatically added to the corresponding ECS entity.
```
Comment (ElleEss) [inline docs/content/component-data.md]: I assume the components belong to the SatialOS entity and the the automatic adding happens as part of the checkout process but this isn't crystal clear if it is. 
Suggest:
"When a SpatialOS entity is [checked out](entity-checkout-process.md), its generated components are automatically added to the corresponding ECS entity as part of the entity check out process . "

File: docs/content/component-data.md [hunk 18]
```diff
@@ -9,56 +9,33 @@ SpatialOS components are generated from `.schema` files into components that the
...
-  public Dictionary<int, float> MapNonBlittable;
-  public string StringNonBlittable;
-}
-```
-
 ### Reading and writing
 
 When a SpatialOS entity is [checked out](entity-checkout-process.md), the generated components will be automatically added to the corresponding ECS entity. 
 
 A component's value can be read by injecting that component into a system, just like any other ECS component.
```
Comment (ElleEss) [inline docs/content/component-data.md]: Does this mean the generated components mentioned in the paragraph above? It's not clear if that is what it means.
Suggest:
"These generated component values can be read by injecting ......"

File: docs/content/component-data.md (hunk 14 above)
Comment (ElleEss) [reply docs/content/component-data.md]: Also:
I think the first sentence is an odd way round. It says "Spatial OS components are generated from .schema files into..."
I don't think anything can "be generated into" <a thing>. You can generate <a thing> and generate from <a thing>. We need to say what causes the generation,..
So:
"<A thing  - The Unity GDK?> uses `.schema` files to generate SpatialOS components into components that the Unity ECS can understand."

File: docs/content/component-data.md [hunk 19]
```diff
@@ -9,56 +9,33 @@ SpatialOS components are generated from `.schema` files into components that the
...
-Note that if the component is non-blittable then it must be injected as a `ComponentArray` rather than a `ComponentDataArray`.
 
 To send a component update, set the component to the value to be sent.
 It will automatically be sent at the end of the tick.
```
Comment (ElleEss) [inline docs/content/component-data.md]: "It is automatically..." 
It definitely happens.

File: docs/content/component-data.md [hunk 20]
```diff
@@ -101,28 +73,28 @@ When a component update is received this will be added as a [reactive component]
 #### Primitive types
 Each primitive type in schemalang corresponds to some type in the Unity GDK.
```
Comment (ElleEss) [inline docs/content/component-data.md]: "....to some type..." -> " ...to a type..."

File: docs/content/create-entity.md [hunk 21]
```diff
@@ -11,37 +11,48 @@ To create an [entity](https://docs.improbable.io/reference/13.0/shared/glossary#
 
 ### 1. Create an entity definition using the `EntityBuilder`
 
-The `EntityBuilder` class is part of the `Improbable.Gdk.Legacy` assembly. It's a convenient way to create entity definitions, and allows you to specify [authority](authority.md) on a per-[component](https://docs.improbable.io/reference/13.0/shared/glossary#component) basis.
+The `EntityBuilder` class is part of the `Improbable.Gdk.Core` assembly. It's a convenient way to create entity definitions, and allows you to specify [authority](authority.md) on a per-[component](https://docs.improbable.io/reference/13.0/shared/glossary#component) basis.
```
Comment (ElleEss) [inline docs/content/create-entity.md]: Comma after "definitions" reads oddly.

Are we pinned to 13.0? (Just checking.)

File: docs/content/create-entity.md [hunk 22]
```diff
@@ -11,37 +11,48 @@ To create an [entity](https://docs.improbable.io/reference/13.0/shared/glossary#
...
+| `SetReadAcl(string attribute, param string[] attribute)` or `SetReadAcl(List<string> attribute)` | Specify which worker type(s) can have [read access](https://docs.improbable.io/reference/13.0/shared/glossary#read-and-write-access-authority) to the entity. |
+| `SetEntityAclComponentWriteAccess(string attribute)` | Specify which worker type can have authority over the [`EntityAcl`](https://docs.improbable.io/reference/13.0/shared/schema/standard-schema-library#entityacl-required) component of your entity. This is useful if, while the game is running, you want to change which worker type(s) can have authority over the entity's other components. |
+| `AddComponent(ComponentData data, string writeAccess)` | Add a user-defined component to your entity and specify which worker type(s) can have authority over it. |
 | `Build()` | Create a finished entity definition. |
 
+For each schema component you define, we generate a helper method to create a `ComponentData` object for that component. For example, for the following schema:
```
Comment (ElleEss) [inline docs/content/create-entity.md]: Who is "we" here?

File: docs/content/create-entity.md [hunk 23]
```diff
@@ -11,37 +11,48 @@ To create an [entity](https://docs.improbable.io/reference/13.0/shared/glossary#
...
+
+```
+component Example {
+  id = 1;
+  int32 value = 1;
+}
+```
+There will be a method `Example.CreateSchemaComponentData(int value)` which will instantiate and populate a `ComponentData` object.
```
Comment (ElleEss) [inline docs/content/create-entity.md]: There will or there is?

File: docs/content/events.md [hunk 24]
```diff
@@ -13,9 +13,9 @@ Events are for broadcasting information between worker instances about a transie
 
 ### Sending events
 
-A worker instance can send an event using a `EventSender<T>` ECS component (where `T` is the SpatialOS component that the event is defined in).
+A worker instance can send an event using a `ComponentName.EventSenders.EventName` ECS component, where `ComponentName` is the name of the component that the event is defined in and `EventName` is the name of the event in schema.
```
Comment (ElleEss) [inline docs/content/events.md]: "..in, and.." 
In this circumstance the "in" followed by the "and" trips you up when you are reading it. A comma will help.

File: docs/content/events.md [hunk 25]
```diff
@@ -13,9 +13,9 @@ Events are for broadcasting information between worker instances about a transie
...
 
-For each SpatialOS component containing an event, the Unity GDK automatically attaches an `EventSender<T>` ECS component when (and only when) the worker instance has authority over the SpatialOS component `T`.
+For each SpatialOS component containing an event, the Unity GDK automatically attaches an `ComponentName.EventSenders.EventName` ECS component when (and only when) the worker instance has authority over the SpatialOS component `ComponentName`.
```
Comment (ElleEss) [inline docs/content/events.md]: "attaches an" -> "attaches a"

File: docs/content/events.md [hunk 26]
```diff
@@ -44,11 +44,11 @@ component CubeColor {
 Given the example schema, the Unity GDK generates these types:
 
 * `ColorData` - Equivalent of the schema type.
-* `ChangeColorEvent` - The corresponding type for the event which contains a `ColorData` within it. This type exists to distinguish between multiple events that use the same type.
+* `CubeColor.EventSenders.ChangeColor` - The event sender type
```
Comment (ElleEss) [inline docs/content/events.md]: ...type.  <- needs a full stop.

File: docs/content/events.md [hunk 27]
```diff
@@ -85,11 +85,9 @@ public class SendChangeColorEvent : ComponentSystem
 
 When a worker instance receives an event, this is represented with reactive ECS components. 
 
-For SpatialOS entity that the event was sent on, the Unity GDK attaches an `EventsReceived<T>` component to the corresponding ECS entity (where `T` is the generated type associated with the event). The ECS component holds a list of `T`, each one being an event.
+For the SpatialOS entity that the event was sent on, the Unity GDK attaches an `ComponentName.ReceivedEvents.EventName` component to the corresponding ECS entity, where `ComponentName` is the name of the component that the event is defined in and `EventName` is the name of the event in schema.
```
Comment (ElleEss) [inline docs/content/events.md]: "attaches an" -> "attaches a"
"defined in and " -> "defined in, and "

File: docs/content/reactive-components.md [hunk 28]
```diff
@@ -13,13 +13,11 @@ At the end of the tick, the Unity GDK removes the reactive component.
...
-
 These are the types of reactive component available:
 
-1. `ComponentsUpdated`:  All local and received [SpatialOS component updates](https://docs.improbable.io/reference/13.0/shared/design/operations#component-related-operations) for the current SpatialOS entity.
-2. `AuthoritiesChanged`: Updates to the [authority](https://docs.improbable.io/reference/13.0/shared/design/understanding-access#understanding-read-and-write-access-authority) the current worker instance has over a SpatialOS component. See [Authority](authority.md) for information on how this works.
-3. `EventsReceived`: All received [events](https://docs.improbable.io/reference/13.0/shared/design/object-interaction#events) for the current entity. See [Events](events.md) for information on how this works.
+1. `ReceivedUpdates`:  All local and received [SpatialOS component updates](https://docs.improbable.io/reference/13.0/shared/design/operations#component-related-operations) for the current SpatialOS entity.
```
Comment (ElleEss) [inline docs/content/reactive-components.md]: Usual question about version pinning.

File: docs/content/commands.md (hunk 10 above)
Comment (jamiebrynes7) [reply docs/content/commands.md]: 1. It doesn't inject anything into the entities, it (Unity's ECS impl) injects the entities into the system . 
2. It sends a `build_wall` command to every entity that the worker can see that has a `Builder` component
3. It actually depends - I'm going to remove this as its dependent on many other factors 
4. Going to drop it. 

Also - the system doesn't specify which worker this runs on, in fact, as its written now it will run on any worker which has an entity checked out with the `Builder` component. Will edit to make it more clear.

Will rephrase as:

"This system is an example of sending command requests. Unity injects all ECS entities which have [list of components] into this system before `OnUpdate`. This system iterates over the injected entities and sends a `build_wall` command request to each of them."

File: docs/content/commands.md (hunk 12 above)
Comment (jamiebrynes7) [reply docs/content/commands.md]: Was erroneous. Removed

File: docs/content/component-data.md (hunk 14 above)
Comment (jamiebrynes7) [reply docs/content/component-data.md]: I don't think explaining the code generation process is in scope for this doc - will link to the existing one. 

Code generation is (now) run automatically when you open Unity or when you press a button in the Editor - will make a note of that

File: docs/content/create-entity.md (hunk 21 above)
Comment (jamiebrynes7) [reply docs/content/create-entity.md]: We are pinned to a version that does not have docs on the website. Will change to latest as that's the closest to the version we have pinned.

File: docs/content/custom-replication-system.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (jamiebrynes7) [reply docs/content/custom-replication-system.md]: https://improbableio.atlassian.net/browse/UTY-1079

File: docs/content/commands.md [hunk 29]
```diff
@@ -70,34 +72,37 @@ public class BuildSystem : ComponentSystem
...
             
-            requestSender.SendBuildWallRequest(entityId, buildRequest);
+            requestSender.RequestsToSend.Add(request);
         }
     }
 }
 ```
 
-This system runs on a client. It injects all SpatialOS entities in the client's view that have the `Builder` SpatialOS component, and sends a `build_wall` command to each SpatialOS entity (which will be received by the managed worker).
+This system is an example of sending command requests. Unity injects all ECS entities which have the `ShouldSendBuildWallCommand`, `Builder.CommandSenders.BuildWall`, and `SpatialEntityId` components into this system before `OnUpdate`. This system iterates over the injected entities and sends a build_wall command request to each of them.
```
Comment (ElleEss) [inline docs/content/commands.md]: `build_wall`
(If we want to be consistent and super clear.)

Comment (jamiebrynes7) [reply docs/content/commands.md]: Woops.. nice catch :)

File: docs/content/commands.md [hunk 30]
```diff
@@ -70,34 +72,37 @@ public class BuildSystem : ComponentSystem
...
 
-To send a `build_wall` command in a system, you need to inject `CommandRequestSender<Builder>` into the system like any other ECS component.
+To send a `build_wall` command in a system, you need to inject `Builder.CommandSenders.BuildWall` into the system like any other ECS component.
 
 ### Responding to command requests
 
 When a worker instance receives a command request, the command request is represented with reactive ECS components. 
 
-The Unity GDK attaches a `CommandRequests<T>` ECS component to the specified ECS entity (where `T` implements `IIncomingCommandRequest`). `CommandRequests<T>` holds a list of `T`. The Unity GDK cleans it up at the end of the tick.
+The Unity GDK attaches a `ComponentName.CommandRequests.CommandName` ECS component to the specified ECS entity, where `ComponentName` is the SpatialOS component the command is defined in and `CommandName` is the name of the command in schema. `ComponentName.CommandRequests.CommandName` contains a list of type `ComponentName.CommandName.ReceivedRequest`. The Unity GDK cleans it up at the end of the tick.
```
Comment (ElleEss) [inline docs/content/commands.md]: This sentence is a bit of a mind-bend.
"The Unity GDK attaches a `ComponentName.CommandRequests.CommandName` ECS component to the specified ECS entity, where `ComponentName` is the SpatialOS component the command is defined in and `CommandName` is the name of the command in schema."

Trying to unpack...maybe split the sentence up with a colon and replace the "and" with a comma:
The Unity GDK attaches a `ComponentName.CommandRequests.CommandName` ECS component to the specified ECS entity: `ComponentName` is the SpatialOS component the command is defined in,  `CommandName` is the name of the command in schema.

File: docs/content/commands.md [hunk 31]
```diff
@@ -117,33 +123,35 @@ public class BuildWallHandlerSystem : ComponentSystem
...
 }
 ```
 
-`build_wall` command requests are on a `CommandRequests<BuildWall.Request>` ECS component. The worker instance should respond to the request using the `SendBuildWallResponse(BuildResponse buildResponse)` method on the request itself.
+`build_wall` command requests are on a `Builder.CommandRequest.BuildWall` ECS component. The worker instance should respond to the request using the `Builder.CommandResponders.BuildWall` ECS component.
 
 ### Receiving command responses
 
-Like requests, when an ECS entity receives a command response, the Unity GDK attaches a `CommandResponses<T>` ECS component to the ECS entity (where `T` implements `IIncomingCommandResponse`).
+Like requests, when an ECS entity receives a command response, the Unity GDK attaches a `ComponentName.CommandResponses.CommandName` ECS component to the ECS entity, where `ComponentName` is the SpatialOS component the command is defined in and `CommandName` is the name of the command in schema.
```
Comment (ElleEss) [inline docs/content/commands.md]: A similar sentence as above but it seems less hard to follow. Go figure?!

File: docs/content/commands.md [hunk 32]
```diff
@@ -117,33 +123,35 @@ public class BuildWallHandlerSystem : ComponentSystem
...
 
-The ECS entity that sent the request receives the response. The response object includes the payload of the response and the request payload that originally send the command. **This payload is null** when no payload is sent back.
+`ComponentName.CommandResponses.CommandName` contains a list of `ComponentName.CommandName.ReceivedResponse`. The `ComponentName.CommandName.ReceivedResponse` includes the payload of the response and the request payload that originally was sent with the command. **This payload is null** when the command fails.
```
Comment (ElleEss) [inline docs/content/commands.md]: "...was originally sent..." reads a bit smoother to me but it's a minor issue.

File: docs/content/commands.md [hunk 33]
```diff
@@ -182,43 +190,37 @@ public class BuildWallResponseHandler : ComponentSystem
...
-* EntityQueryRequest
-    * Method - `SendEntityQueryRequest(Worker.Query.EntityQuery entityQuery, uint timeoutMillis = 0)`
-
-        For more information, see [entity queries](https://docs.improbable.io/reference/13.0/shared/glossary#queries) in the SpatialOS documentation.
-
-        `timeoutMillis` is optional. 
-    * Response - `EntityQueryResponse`
-
-When a response is received, the Unity GDK attaches a `CommandResponses<T>` ECS component to the ECS entity that originally sent the request. `T` is the corresponding response type.
+When a response is received, the Unity GDK attaches the corresponding responses ECS component to the ECS entity that originally sent the request.
```
Comment (ElleEss) [inline docs/content/commands.md]: "When a response is received, the Unity GDK attaches the corresponding responses ECS component to the ECS entity that originally sent the request. "
This is a bit confusing. I think it should be:
"When a response is received, the Unity GDK attaches an ECS component to the ECS entity that originally sent the request. The ECS component it attaches corresponds to the response it receives."

File: docs/content/component-data.md [hunk 34]
```diff
@@ -4,64 +4,41 @@
 
 
 ## Component Data
-SpatialOS components are generated from `.schema` files into components that the Unity ECS can understand. See the schemalang [docs](https://docs.improbable.io/reference/13.0/shared/schema/introduction#schema-introduction) for details on how to create schema components.
+The [code generator](./code-generator.md) uses `.schema` files to generate components that the Unity ECS can understand. See the schemalang [docs](https://docs.improbable.io/reference/latest/shared/schema/introduction#schema-introduction) for details on how to create schema components.
+
+> Note that code generation is run when you open Unity or by pressing Improbable > Generate Code in the Unity menu.
```
Comment (ElleEss) [inline docs/content/component-data.md]: Nice. :-)

File: docs/content/component-data.md (hunk 14 above)
Comment (ElleEss) [reply docs/content/component-data.md]: Just that addition above with and small introduction and when it happens is 100% clearer. <3 it.

File: docs/content/custom-replication-system.md [hunk 35]
```diff
@@ -20,7 +20,7 @@ When a worker sends a SpatialOS event, the Unity GDK puts the event object into
 
 If some ECS components need more complex replication logic, you can create custom replication systems on a per-component basis. To do this:
 
-* Your custom replication system must extend the `CustomSpatialOSSendSystem<T>` class (where `T` is a SpatialOS component). Note that this will disable the standard replication for `T`.
+* Your custom replication system must extend the `Improbable.Gdk.Core.CustomSpatialOSSendSystem<T>` class (where `T` is a SpatialOS component). Note that this will disable the standard replication for `T` and ensure the system runs at the correct point in the update lifecycle. For more information about the update lifecycle see [System Update Order](./system-update-order.md).
```
Comment (ElleEss) [inline docs/content/custom-replication-system.md]: "...this disables..."
(If you are changing anything else, it's good to get rid of the prophecies. )
